### PR TITLE
Stabilize default HikariCP pool metric names

### PR DESCRIPTION
**Pull request by YaoYingLong**
## Changes

- Derive stable metric pool names from JDBC connection information when `poolName` is not configured.
- Fall back to `hikaricp` when connection information is unavailable.
- Preserve explicit HikariCP pool names, JMX names, and names passed to custom metrics trackers.
- Add regression tests for copied configs, repeated validation, duplicate pools, and stable semconv.

### CODE REVIEW

File: instrumentation/hikaricp-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/hikaricp/v3_0/HikariSingletons.java [hunk 1]
```diff
@@ -5,20 +5,50 @@
...
+import io.opentelemetry.instrumentation.jdbc.internal.JdbcConnectionPoolNameUtil;
+import io.opentelemetry.instrumentation.jdbc.internal.JdbcConnectionUrlParser;
+import io.opentelemetry.javaagent.bootstrap.jdbc.DbInfo;
 import javax.annotation.Nullable;
 
 public class HikariSingletons {
 
+  private static final String DEFAULT_DATA_SOURCE_NAME = "hikaricp";
+
+  private static final VirtualField<HikariConfig, Boolean> generatedPoolNameField =
```
Comment (trask) [inline instrumentation/hikaricp-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/hikaricp/v3_0/HikariSingletons.java]: [Style] `VirtualField` handles count as constant-like values in the style guide, so this field should be `SCREAMING_SNAKE_CASE`. Renaming it to `GENERATED_POOL_NAME_FIELD` (and updating the three usages below) would match how the other instrumentation modules declare their virtual fields.

File: instrumentation/hikaricp-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/hikaricp/v3_0/HikariSingletons.java [hunk 2]
```diff
@@ -5,20 +5,50 @@
...
+    if (Boolean.TRUE.equals(generatedPoolNameField.get(config))) {
+      return hikariTelemetry.createMetricsTrackerFactory(delegate, getDataSourceName(config));
+    }
     return hikariTelemetry.createMetricsTrackerFactory(delegate);
   }
 
+  private static String getDataSourceName(HikariConfig config) {
+    DbInfo dbInfo =
+        JdbcConnectionUrlParser.parse(config.getJdbcUrl(), config.getDataSourceProperties());
+    return JdbcConnectionPoolNameUtil.poolName(dbInfo, DEFAULT_DATA_SOURCE_NAME);
```
Comment (trask) [inline instrumentation/hikaricp-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/hikaricp/v3_0/HikariSingletons.java:63]: [General] In HikariCP's `dataSourceClassName` configuration mode there is normally no `jdbcUrl` at all (Hikari warns and ignores one if it is set), so `getJdbcUrl()` is null here. `JdbcConnectionUrlParser.parse()` returns `DbInfo.DEFAULT` immediately for a null URL without ever consulting `props`, so the standard `serverName`/`portNumber`/`databaseName` entries in `getDataSourceProperties()` are dropped and every such pool derives `hikaricp`. The derived name is then not just uninformative but shared: those pools report distinct `HikariPool-N` values in `pool.name` today, and afterwards they would all report `hikaricp`. Falling back to the standard data source properties when the URL is absent would close the gap.

`shouldUseDataSourcePropertiesForDefaultPoolName` doesn't cover this, since it injects a `DataSource` instance and supplies a synthetic `jdbc:postgresql:ignored` URL — HikariCP ignores `dataSourceProperties` entirely in that branch.

**Review by trask — COMMENTED** [review]
Lightly filtered AI-generated feedback

File: instrumentation/hikaricp-3.0/library/src/main/java/io/opentelemetry/instrumentation/hikaricp/v3_0/HikariTelemetry.java [hunk 3]
```diff
@@ -39,6 +39,16 @@ public MetricsTrackerFactory createMetricsTrackerFactory() {
...
-    return new OpenTelemetryMetricsTrackerFactory(openTelemetry, delegate);
+    return new OpenTelemetryMetricsTrackerFactory(openTelemetry, delegate, null);
+  }
+
+  /**
+   * Returns a new {@link MetricsTrackerFactory} that uses {@code poolNameOverride} for
+   * OpenTelemetry metrics. Trackers created by the {@code delegate} metrics tracker factory will
+   * continue to use the pool name provided by HikariCP.
+   */
+  public MetricsTrackerFactory createMetricsTrackerFactory(
```
Comment (trask) [inline instrumentation/hikaricp-3.0/library/src/main/java/io/opentelemetry/instrumentation/hikaricp/v3_0/HikariTelemetry.java]: Does this need to be library API at all? A library user sets the pool name themselves via `config.setPoolName(...)`, so `poolNameOverride` only exists to serve the javaagent.

Since `MetricsTrackerFactory` is a functional interface and `PoolBase.setMetricsTrackerFactory` calls `create(config.getPoolName(), stats)` synchronously, the agent can swap the name on the way in and swap it back for the user's delegate, leaving both library classes unchanged:

```java
public static MetricsTrackerFactory createMetricsTrackerFactory(
    @Nullable MetricsTrackerFactory delegate, HikariConfig config) {
  if (!Boolean.TRUE.equals(GENERATED_POOL_NAME_FIELD.get(config))) {
    return hikariTelemetry.createMetricsTrackerFactory(delegate);
  }
  String dataSourceName = getDataSourceName(config);
  return (hikariPoolName, poolStats) ->
      hikariTelemetry
          .createMetricsTrackerFactory(
              delegate == null ? null : (ignored, stats) -> delegate.create(hikariPoolName, stats))
          .create(dataSourceName, poolStats);
}
```

If you prefer keeping it in the library module, `...hikaricp.v3_0.internal` with the standard "internal and is hence not for public use" javadoc would keep it off the published surface.

**Review by trask — APPROVED** [review]
